front/pages/[user]/a/[sId]/datasets/[name]/index.jsx

```diff
@@ -23,12 +24,35 @@ export default function ViewDatasetView({
 
   const [disable, setDisabled] = useState(true);
   const [loading, setLoading] = useState(false);
+  const [editorDirty, setEditorDirty] = useState(false);
+  const [isFinishedEditing, setIsFinishedEditing] = useState(false);
   const [updatedDataset, setUpdatedDataset] = useState(dataset);
 
-  const onUpdate = (valid, dataset) => {
+  useRegisterUnloadHandlers(editorDirty);
+
+  // this is a little wonky, but in order to redirect to the datasets main page and not
```

Comment:
We try to have comments presented as full sentence: aka capitalized at the beginning and finished with a dot (already the case)